consensus/propagation/reactor.go

```diff
@@ -465,14 +475,61 @@ func (blockProp *Reactor) handleWants(peer p2p.ID, wants *proptypes.WantParts) {
 	_, parts, _, has := blockProp.GetProposal(height, round)
 	// the peer must always send the proposal before sending parts, if they did
 	//  not, this node must disconnect from them.
-	if !has {
+	// TODO fix this hack where we check for round -2
+	if round != -2 && !has {
 		blockProp.Logger.Error("received part state request for unknown proposal", "peer", peer, "height", height, "round", round)
 		// d.pswitch.StopPeerForError(p.peer, fmt.Errorf("received part state for unknown proposal"))
 		return
 	}
 
 	// send the peer the partset header if they don't have the proposal.
 	// TODO get rid of this catchup case
+	if round == -2 {
```

Comment:
per some previous discussions I think we can just create new message(s) for catchup. I think we'll end up having to do it no matter what, but it also open the design space in the future without being quite as breaking.